### PR TITLE
Test improvement

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Add `php-7.4` test on Travis CI build.
- Update the cached dependencies to let `php-7.1` version can install with `composer.lock` file.
- The `composer.lock` has been under Git version control, and it should use `composer install` command during Travis CI build.
- Add `DivineOmega\\FuzzyEvents\\Unit\\` namesapce on `tests/Unit` folder to avoid `The use statement with non-compound name 'InvalidArgumentException' has no effect` warning.
- Add `testEmptyListener` test.
- Removing unnecessary variables. It's defined, but not used.
- Using the `assertInstanceOf` assertion to assert expected class instance is same as result.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,10 @@ php:
   - '7.1'
   - '7.2'
   - '7.3'
+  - '7.4'
 install:
-  - composer update
+  - composer install
```
Comment (Jord-JD) [inline .travis.yml:9]: It was a mistake that the `composer.lock` file was committed. Could you remove `composer.lock` and change this back to `composer update`.

**Review by Jord-JD — CHANGES_REQUESTED** [review]
Thanks. Just one change please.

File: .travis.yml (hunk 1 above)
Comment (Jord-JD) [reply .travis.yml:9]: Actually, it's fine to keep it as `composer install`, but `composer.lock` should not be there.

Comment (peter279k) [reply .travis.yml:9]: I think it should remove `composer.lock` and keep `composer install` command existing :).

**Review by Jord-JD — APPROVED** [review]

**Comment by Jord-JD** [comment]
Great. Thank you.

**Pull request merged**